### DISCUSSION TITLE
Receiving P2P messages with Crazyradio

### DISCUSSION THREAD

**Discussion by williamleong** [bitcraze/discussions#1133] [Q&A] [answered]
Has there been any updates since https://github.com/bitcraze/crazyflie-lib-python/issues/147 to allow P2P messages to be received by Crazyradio? We would like to send some custom detection data from the Crazyflies on demand to be processed using P2P broadcast and treat the computer as another peer in the system.

**Comment by ataffanel** [comment, marked as answer]
No, unfortunately there has not been any progress so far on this front. We hope to be able to start working on that in the following month.

In the mean-time the quickest option is still to use a Crazyflie as a bridge: write an app in the Crazyflie that receive data and makes them available as either Log variables or an app channel, and connect the Crazyflie using USB from the PC side to collect these information. This is how we have been setting up our decentralized experiments and demo.

**Reply by williamleong** [reply]
Thanks, I have done a bit of work with the assumption that I would be able to use the Crazyradio for receiving P2P messages as I was able to send such messages. I will try the methods you suggested in the meantime as a workaround.

**Comment by r-bt** [comment]
@ataffanel any update on this?

**Reply by ArisMorgens** [reply]
Hello @r-bt!

Unfortunately no, there has been no progress on P2P communication with Crazyradio 2.0.

For our [decentralized swarm demo](https://www.bitcraze.io/2025/03/decentralized-brushless-swarm-demo/), we're still using a non-flying Crazyflie as the "sniffer" which is connected to the PC via usb. You can find the P2P firmware for this demo in our [crazyflie-firmware-experimental](https://github.com/bitcraze/crazyflie-firmware-experimental/tree/master/examples/demos/decentralized_swarm) repo.

Let us know if you need anything else.

**Reply by r-bt** [reply]
Hi @ArisMorgens, is there any update on when this will be supported? The decentralized swarm demo works great however it's annoying to have to give up an entire drone to control it.

**Reply by ArisMorgens** [reply]
@Hi @r-bt !

I agree that giving up a Crazyflie is inconvenient. P2P communication via Crazyradio is a great feature to have but other radio priorities are ahead of it, so I can't provide a specific timeline.
